### ISSUE TITLE
Push panel

### ISSUE THREAD

**Issue by petervachon** [carbon-design-system/ibm-security#639]
## Request 
We are looking to explore options for our panels, both right panel, and shell navigation to have a push state, which would effectively push the content to allow for a user to view panel information in parallel with the page content.

### Details
Hoping the POC would outline Pros & Cons for implementing such a feature including performance and user experience. This is an older prototype but should hopefully illustrate the type of behavior we are looking for: https://ibm.box.com/s/ei4ernt1gejdh2b6ujmdmckopm0dx3i8

**Comment by SimonFinney** [comment]
Superseded by https://github.com/carbon-design-system/ibm-cloud-cognitive/issues/64
